packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3120,7 +3127,7 @@ export class ContainerRuntime
 		} finally {
 			this._orderSequentiallyCalls--;
 		}
-
+		stageControls?.commitChanges();
 		// We don't flush on TurnBased since we expect all messages in the same JS turn to be part of the same batch
 		if (this.flushMode !== FlushMode.TurnBased && this._orderSequentiallyCalls === 0) {
```

Comment:
Btw you could just call `!this.currentlyBatching`, not sure if that's clearer or not though :P

Reply:
i actually don't think that is quite right. i think that would flush after every nested call, not just the top call